compiler/qsc_passes/src/measurement.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod tests;
+
+use miette::Diagnostic;
+use qsc_data_structures::span::Span;
+use qsc_hir::{
+    hir::{Attr, CallableDecl, Item, ItemKind, Package, SpecBody, SpecGen},
+    ty::{Prim, Ty},
+    visit::Visitor,
+};
+use thiserror::Error;
+
+#[derive(Clone, Debug, Diagnostic, Error)]
+pub enum Error {
+    #[error("a measurement should take at least one Qubit as argument")]
+    #[diagnostic(code("Qsc.Measurement.NoArguments"))]
+    NoArguments(#[label] Span),
+
+    #[error("a measurement should only take Qubits as arguments")]
```

Comment:
Is this inherently true? Is it not possible someone could provide a measurement for example that allows you to specify an arbitrary basis as a set of vectors or some other parameterized measurement? (Fine if this is just a restriction that's reasonable for now. Just want to understand).